flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolImpl.java

```diff
@@ -447,12 +452,40 @@ public void releaseSlot(@Nonnull SlotRequestId slotRequestId, @Nullable Throwabl
 
 		componentMainThreadExecutor.assertRunningInMainThread();
 
+		checkState(batchSlotTimeoutCheckEnabled, "batch slot timeout check is disabled unexpectedly.");
+
 		final PendingRequest pendingRequest = PendingRequest.createBatchRequest(slotRequestId, resourceProfile);
 
 		return requestNewAllocatedSlotInternal(pendingRequest)
 			.thenApply(Function.identity());
 	}
 
+	@Override
+	public CompletableFuture<PhysicalSlot> requestNewAllocatedSlotWithoutTimeout(
+			final SlotRequestId slotRequestId,
+			final ResourceProfile resourceProfile,
+			final boolean isBatchRequest) {
+
+		componentMainThreadExecutor.assertRunningInMainThread();
+
+		// a slot pool can serve either this kind of request or requestNewAllocatedBatchSlot(...), never both
+		disableBatchSlotTimeoutCheck();
```

Comment:
```suggestion
		batchSlotTimeoutCheckEnabled = false;
```
do we need a method for this?

Reply:
we can remove this method.

Reply:
no more `requestNewAllocatedSlotWithoutTimeout()` anymore so no change is needed.